tinygrad/codegen/uops.py

```diff
@@ -25,6 +25,29 @@ class UOp:
   def __repr__(self):
     return f"{str(self.uop):20s}: {str(self.dtype) if self.dtype is not None else '':25s} {str([x.uop for x in self.vin]):32s} {self.arg}"
 
+def exec_alu(arg, dtype, p):
+  if arg == TernaryOps.WHERE: ret = p[1] if p[0] else p[2]
+  elif arg == UnaryOps.LOG2: ret = math.log2(p[0]) if p[0] > 0 else -math.inf if p[0] == 0 else math.nan
+  elif arg == UnaryOps.EXP2:
+    try: ret = math.exp(p[0]*math.log(2))
+    except OverflowError: ret = math.inf
+  elif arg == UnaryOps.SQRT: ret = math.sqrt(p[0]) if p[0] >= 0 else math.nan
+  elif arg == UnaryOps.SIN: ret = math.sin(p[0])
+  elif arg == UnaryOps.NEG: ret = -p[0]
+  elif arg == BinaryOps.MUL: ret = p[0]*p[1]
+  elif arg == BinaryOps.ADD: ret = p[0]+p[1]
+  elif arg == BinaryOps.SUB: ret = p[0]-p[1]
+  elif arg == BinaryOps.XOR: ret = p[0]^p[1]
+  elif arg == BinaryOps.MAX: ret = max(p[0], p[1])
+  elif arg == BinaryOps.CMPEQ: ret = p[0] == p[1]
+  elif arg == BinaryOps.CMPLT: ret = p[0] < p[1]
+  elif arg == BinaryOps.DIV: ret = p[0]//p[1] if dtypes.is_int(dtype) else (p[0]/p[1] if p[1] != 0 else math.nan)
```

Comment:
@carsonRadtke what if it's int and p[1] is 0?

Reply:
Perhaps we should just let Python raise a ZeroDivisionError? (Or raise it ourselves so it is a bit cleaner)

The other option is to mimic the non-int branch and return math.nan. This feels a bit weird to me because
```
>>> type(math.nan)
<class 'float'>
```